api/src/org/labkey/api/data/ContainerManager.java

```diff
@@ -1741,25 +1741,56 @@ public static boolean isDeleting(final Container c)
         return deletingContainers.containsKey(c.getId());
     }
 
-    // Delete a container from the database
-    public static boolean delete(final Container c, User user, @Nullable String comment)
+    // Delete containers from the database
+    private static boolean delete(final Collection<Container> containers, User user, @Nullable String comment)
     {
-        if (!isDeletable(c))
+        List<String> containerIds = containers.stream().map(Container::getId).toList();
+
+        try
         {
-            throw new IllegalArgumentException("Cannot delete container: " + c.getPath());
+            synchronized (deletingContainers)
+            {
+                for (Container container : containers)
+                {
+                    if (!isDeletable(container))
+                    {
+                        throw new IllegalArgumentException("Cannot delete container: " + container.getPath());
+                    }
+
+                    if (deletingContainers.containsKey(container.getId()))
+                    {
+                        throw new IllegalArgumentException("Container is already being deleted: " + container.getPath());
+                    }
+                }
+                containerIds.forEach(id -> deletingContainers.put(id, user.getUserId()));
+            }
+            List<Boolean> deleted = new ArrayList<>();
+            for (Container c : containers)
+            {
+                Boolean delete = delete(c, user, comment);
+                deleted.add(delete);
+            }
+            return deleted.stream().allMatch(Boolean::booleanValue);

```

Comment:
There's no need to keep the list of true/false.

```suggestion
            boolean deleted = true;
            for (Container c : containers)
            {
                deleted = deleted && delete(c, user, comment);
            }
            return deleted;
```